# COMPLEXIDADE ECONÔMICA E COVID-19: UMA BREVE ANÁLISE DA CAPACIDADE DE RESPOSTA DOS MUNICÍPIOS DO ESPÍRITO SANTO

# 1.Importação da base de dados

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
!pip install kaleido

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 79.9/79.9 MB 8.3 MB/s eta 0:00:00


In [ ]:
import kaleido
import pandas as pd

In [ ]:
import plotly.express as px

In [ ]:
from google.colab import drive
drive.mount('/content/drive/')

Mounted at /content/drive/


In [ ]:
df = pd.read_excel('/content/drive/My Drive/TRABALHO/BaseOficial.xlsx')

In [ ]:
df

CODMUN   SH4   ANO       MUNICIPIO  \
0      3200102   901  2015  AFONSO CLAUDIO   
1      3200102  1507  2017  AFONSO CLAUDIO   
2      3200102   901  2018  AFONSO CLAUDIO   
3      3200102  1201  2018  AFONSO CLAUDIO   
4      3200102   901  2019  AFONSO CLAUDIO   
...        ...   ...   ...             ...   
10464  3205309  9617  2022         VITORIA   
10465  3205309  9618  2022         VITORIA   
10466  3205309  9701  2022         VITORIA   
10467  3205309  9702  2022         VITORIA   
10468  3205309  9703  2022         VITORIA   

                                              PRODUTOSH4  \
0      Café, mesmo torrado ou descafeinado; cascas e ...   
1      Óleo de soja e respectivas fracções, mesmo ref...   
2      Café, mesmo torrado ou descafeinado; cascas e ...   
3                                  Soja, mesmo triturada   
4      Café, mesmo torrado ou descafeinado; cascas e ...   
...                                                  ...   
10464  Garrafas térmicas e outros recipientes isotérm...   
10465  Manequins e artigos semelhantes; autômatos e c...   
10466  Quadros, pinturas e desenhos, feitos inteirame...   
10467        Gravuras, estampas e litografias, originais   
10468  Produções originais de arte estatuária ou de e...   

                                              PRODUTOSH2  \
0                          Café, chá, mate e especiarias   
1      Gorduras e óleos animais ou vegetais; produtos...   
2                          Café, chá, mate e especiarias   
3      Sementes e frutos oleaginosos; grãos, sementes...   
4                          Café, chá, mate e especiarias   
...                                                  ...   
10464                                     Obras diversas   
10465                                     Obras diversas   
10466         Objetos de arte, de coleção e antiguidades   
10467         Objetos de arte, de coleção e antiguidades   
10468         Objetos de arte, de coleção e antiguidades   

              CLASSIFICACAO      EXP       ECI  VCR  ... OBITO ISOLAMENTO  \
0        Produtos Primários  2983651 -0.289533    1  ...     3          1   
1      Baseados em Recursos   465320  0.021937    1  ...     3          1   
2        Produtos Primários    11500 -0.021426    1  ...     3          1   
3        Produtos Primários   435598 -0.021426    1  ...     3          1   
4        Produtos Primários   250621 -0.511772    1  ...     3          1   
...                     ...      ...       ...  ...  ...   ...        ...   
10464      Baixa Tecnologia      248  0.020391    0  ...   722          1   
10465      Baixa Tecnologia      295  0.020391    0  ...   722          1   
10466                Outros    12530  0.020391    0  ...   722          1   
10467                Outros        7  0.020391    0  ...   722          1   
10468                Outros     1350  0.020391    0  ...   722          1   

       MONITORAMENTO  BARREIRAS SANCOES  TENDAS  SOBRECARGA  AMPLIACAOLEITE  \
0                  0          1       1       1           0               1   
1                  0          1       1       1           0               1   
2                  0          1       1       1           0               1   
3                  0          1       1       1           0               1   
4                  0          1       1       1           0               1   
...              ...        ...     ...     ...         ...             ...   
10464              1          1       1       0           0               0   
10465              1          1       1       0           0               0   
10466              1          1       1       0           0               0   
10467              1          1       1       0           0               0   
10468              1          1       1       0           0               0   

       HOSPITALCAMPANHA         FAIXA_POP_2020  
0                     0    4 - 20001 até 50000  
1                     0    4 - 20001 até 

# 2.Análise inicial do dataframe

## 2.1.Tipo das variáveis

Estamos trabalhando com um dataframe composto por 21 variáveis:
- CODMUN: código munícipio no IBGE.
- SH4: código de 4 dígitos do produto no Sistema Harmonizado.
- ANO: ano da exportação.
- MUNICÍPIO: nome do município brasileiro.
- PRODUTOSH4: descrição da classificação de 4 dígitos do produto.
- PRODUTOSH2: descrição do grupo (classificação de 2 dígitos) do produto.
- CLASSIFICAÇÃO: categoria de intensidade tecnológica do produto (produto primário, baseado em recurso, baixa tecnologia, média tecnologia, alta tecnologia e outros), de acordo com Lall (2000).
- EXP: valor exportado em reais (valores correntes).
- ECI: índice de complexidade econômica do município.
- VCR: variável binária indicando se o município é exportador competitivo do bem (1) ou não (0).
- PIBpc: PIB per capita em reais (valores correntes).
- CresPIBpc: crescimento do PIB per capita em reais (valores correntes).
- ISOLAMENTO: variável binária indicando se o munícipio adotou alguma medida de isolamento social (1, caso positivo; 0, caso contrário).
- OBITOS: número de mortes por COVID-19 entre março/20 e dezembro/20.
- MONITORAMENTO: variável binária indicando se o munícipio adotou um sistema de monitoramento para avaliar a eficácia do isolamento social (1, caso positivo; 0, caso contrário).
- BARREIRAS: variável binária indicando se o munícipio adotou alguma barreira sanitária nas entradas da cidade (1, caso positivo; 0, caso contrário).
- SANCOES: variável binária indicando se o munícipio adotou alguma medida para regulamentar sanções (1, caso positivo; 0, caso contrário).
- TENDAS: variável binária indicando se houve instalação de tendas de triagem (1, caso positivo; 0, caso contrário).
- SOBRECARGA: variável binária indicando se o número de internações ultrapassou a capacidade de leitos e de unidades de tratamento intensivo (UTI) públicos ou privados conveniados ao SUS (1, caso positivo; 0, caso contrário).
- AMPLIACAOLEITOS: variável binária indicando se o número de leitos foi ampliado para atender à demanda por internação no município (1, caso positivo; 0, caso contrário).
- HOSPITALCAMPANHA: variável binária indicando se foi instalado hospital de campanha durante a pandemia da COVID-19 no município (1, caso positivo; 0, caso contrário).

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10469 entries, 0 to 10468
Data columns (total 22 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   CODMUN            10469 non-null  int64  
 1   SH4               10469 non-null  int64  
 2   ANO               10469 non-null  int64  
 3   MUNICIPIO         10469 non-null  object 
 4   PRODUTOSH4        10469 non-null  object 
 5   PRODUTOSH2        10469 non-null  object 
 6   CLASSIFICACAO     10469 non-null  object 
 7   EXP               10469 non-null  int64  
 8   ECI               10469 non-null  float64
 9   VCR               10469 non-null  int64  
 10  PIBpc             10469 non-null  object 
 11  CresPIBpc         9854 non-null   object 
 12  OBITO             10469 non-null  int64  
 13  ISOLAMENTO        10469 non-null  int64  
 14  MONITORAMENTO     10469 non-null  object 
 15  BARREIRAS         10469 non-null  int64  
 16  SANCOES           10469 non-null  int64 

## 2.2.Transformação das variáveis

Temos que transformar algumas variáveis em *string*, sejam elas códigos ou descrição de alguma informação.

In [ ]:
df['CODMUN'] = df['CODMUN'].astype(str)
df['SH4'] = df['SH4'].astype(str)
df['MUNICIPIO'] = df['MUNICIPIO'].astype(str)
df['PRODUTOSH4'] = df['PRODUTOSH4'].astype(str)
df['PRODUTOSH2'] = df['PRODUTOSH2'].astype(str)
df['CLASSIFICACAO'] = df['CLASSIFICACAO'].astype(str)

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10469 entries, 0 to 10468
Data columns (total 22 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   CODMUN            10469 non-null  object 
 1   SH4               10469 non-null  object 
 2   ANO               10469 non-null  int64  
 3   MUNICIPIO         10469 non-null  object 
 4   PRODUTOSH4        10469 non-null  object 
 5   PRODUTOSH2        10469 non-null  object 
 6   CLASSIFICACAO     10469 non-null  object 
 7   EXP               10469 non-null  int64  
 8   ECI               10469 non-null  float64
 9   VCR               10469 non-null  int64  
 10  PIBpc             10469 non-null  object 
 11  CresPIBpc         9854 non-null   object 
 12  OBITO             10469 non-null  int64  
 13  ISOLAMENTO        10469 non-null  int64  
 14  MONITORAMENTO     10469 non-null  object 
 15  BARREIRAS         10469 non-null  int64  
 16  SANCOES           10469 non-null  int64 

# 3.Limpeza dataframe inicial e organização dados análise

## 3.1.Limpeza dados PIB per capita, crescimento PIB per capita e dummy de monitoramento

Precisamos tratar as informações de PIB per capita, Crescimento Anual do PIB per capita e da Dummy de Monitoramento, pois, em tese, deveriam ser variáveis inteiras ou decimais, mas retornaram como *string*. Provavelmente, existem valores nulos que foram substituídos por texto para indicar a ausência de dados.

## 3.1.1.PIB per capita

In [ ]:
df_analise_missing_pibpc = df[(df['PIBpc'] == 'NÃO DISPONÍVEL')].copy()
df_analise_missing_pibpc.groupby(by=['ANO']).count()

CODMUN   SH4  MUNICIPIO  PRODUTOSH4  PRODUTOSH2  CLASSIFICACAO   EXP  \
ANO                                                                          
2021    1817  1817       1817        1817        1817           1817  1817   
2022    1872  1872       1872        1872        1872           1872  1872   

       ECI   VCR  PIBpc  ...  OBITO  ISOLAMENTO  MONITORAMENTO  BARREIRAS  \
ANO                      ...                                                
2021  1817  1817   1817  ...   1817        1817           1817       1817   
2022  1872  1872   1872  ...   1872        1872           1872       1872   

      SANCOES  TENDAS  SOBRECARGA  AMPLIACAOLEITE  HOSPITALCAMPANHA  \
ANO                                                                   
2021     1817    1817        1817            1817              1817   
2022     1872    1872        1872            1872              1872   

      FAIXA_POP_2020  
ANO                   
2021            1817  
2022            1872  

[2 rows x 21 columns]

Vemos que a variável de PIBpc não possui dados para os anos de 2021 e 2022. Isto ocorre pois o IBGE ainda não estruturou os dados para esses anos, sendo que, na nossa análise, tanto o número de óbitos, quanto a exportação e a complexidade são informações que se estendem até 2022. Dessa forma, criaremos um dataframe separado para a análise do PIB per capita.

In [ ]:
df_analise_pibpc = df[(df['PIBpc'] != 'NÃO DISPONÍVEL')].copy()
df_analise_pibpc['PIBpc'] = df_analise_pibpc['PIBpc'].astype(float)
df_analise_pibpc['CresPIBpc'] = df_analise_pibpc['CresPIBpc'].astype(float)
df_analise_pibpc

CODMUN   SH4   ANO       MUNICIPIO  \
0     3200102   901  2015  AFONSO CLAUDIO   
1     3200102  1507  2017  AFONSO CLAUDIO   
2     3200102   901  2018  AFONSO CLAUDIO   
3     3200102  1201  2018  AFONSO CLAUDIO   
4     3200102   901  2019  AFONSO CLAUDIO   
...       ...   ...   ...             ...   
9441  3205309  9613  2020         VITORIA   
9442  3205309  9614  2020         VITORIA   
9443  3205309  9616  2020         VITORIA   
9444  3205309  9617  2020         VITORIA   
9445  3205309  9618  2020         VITORIA   

                                             PRODUTOSH4  \
0     Café, mesmo torrado ou descafeinado; cascas e ...   
1     Óleo de soja e respectivas fracções, mesmo ref...   
2     Café, mesmo torrado ou descafeinado; cascas e ...   
3                                 Soja, mesmo triturada   
4     Café, mesmo torrado ou descafeinado; cascas e ...   
...                                                 ...   
9441  Isqueiros e outros acendedores (exceto os da p...   
9442  Cachimbos (incluindo os seus fornilhos), pitei...   
9443  Vaporizadores de toucador, suas armações e cab...   
9444  Garrafas térmicas e outros recipientes isotérm...   
9445  Manequins e artigos semelhantes; autômatos e c...   

                                             PRODUTOSH2         CLASSIFICACAO  \
0                         Café, chá, mate e especiarias    Produtos Primários   
1     Gorduras e óleos animais ou vegetais; produtos...  Baseados em Recursos   
2                         Café, chá, mate e especiarias    Produtos Primários   
3     Sementes e frutos oleaginosos; grãos, sementes...    Produtos Primários   
4                         Café, chá, mate e especiarias    Produtos Primários   
...                                                 ...                   ...   
9441                                     Obras diversas      Baixa Tecnologia   
9442                                     Obras diversas      Baixa Tecnologia   
9443                                     Obras diversas      Baixa Tecnologia   
9444                                     Obras diversas      Baixa Tecnologia   
9445                                     Obras diversas      Baixa Tecnologia   

          EXP       ECI  VCR  ...  OBITO  ISOLAMENTO  MONITORAMENTO  \
0     2983651 -0.289533    1  ...      3           1              0   
1      465320  0.021937    1  ...      3           1              0   
2       11500 -0.021426    1  ...      3           1              0   
3      435598 -0.021426    1  ...      3           1              0   
4      250621 -0.511772    1  ...      3           1              0   
...       ...       ...  ...  ...    ...         ...            ...   
9441      332  0.020993    0  ...    722           1              1   
9442       45  0.020993    0  ...    722           1              1   
9443      230  0.020993    0  ...    722           1              1   
9444       95  0.020993    0  ...    722           1              1   
9445      375  0.020993    0  ...    722           1              1   

      BARREIRAS SANCOES  TENDAS  SOBRECARGA  AMPLIACAOLEITE  HOSPITALCAMPANHA  \
0             1       1       1           0               1                 0   
1             1       1       1           0               1                 0   
2             1       1       1           0               1                 0   
3             1       1       1           0               1                 0   
4             1       1       1           0               1                 0   
...         ...     ...     ...         ...             ...               ...   
9441          1       1       0           0               0                 0   
9442          1       1       0           0               0                 0   
9443          1       1       0           0               0                 0   
9444          1       1       0           0               0                 0   
9445          1       1       0         

## 3.1.2.Crescimento do PIB per capita

Vemos abaixo que o ano de 2015 concentra todas as observações nulas de crescimento do PIB per capita da base. Para o primeiro registro de cada município (primeiro ano disponível é 2015) não existe comparação do ano anterior na base. Dessa forma, criaremos um dataframe separado para a análise do Crescimento do PIB per capita.

Como PIB per capita e Crescimento do PIB per capita caminham juntos, iremos excluir os anos de 2015, 2021 e 2022. O objetivo é trabalhar com uma base mais balanceada.

In [ ]:
df.CresPIBpc.isnull().groupby([df['ANO']]).sum().astype(int).reset_index(name='count')

ANO  count
0  2015    615
1  2016      0
2  2017      0
3  2018      0
4  2019      0
5  2020      0
6  2021      0
7  2022      0

In [ ]:
df_analise = df_analise_pibpc[(df_analise_pibpc['ANO'] != 2015)].copy()
df_analise

CODMUN   SH4   ANO       MUNICIPIO  \
1     3200102  1507  2017  AFONSO CLAUDIO   
2     3200102   901  2018  AFONSO CLAUDIO   
3     3200102  1201  2018  AFONSO CLAUDIO   
4     3200102   901  2019  AFONSO CLAUDIO   
5     3200102   901  2020  AFONSO CLAUDIO   
...       ...   ...   ...             ...   
9441  3205309  9613  2020         VITORIA   
9442  3205309  9614  2020         VITORIA   
9443  3205309  9616  2020         VITORIA   
9444  3205309  9617  2020         VITORIA   
9445  3205309  9618  2020         VITORIA   

                                             PRODUTOSH4  \
1     Óleo de soja e respectivas fracções, mesmo ref...   
2     Café, mesmo torrado ou descafeinado; cascas e ...   
3                                 Soja, mesmo triturada   
4     Café, mesmo torrado ou descafeinado; cascas e ...   
5     Café, mesmo torrado ou descafeinado; cascas e ...   
...                                                 ...   
9441  Isqueiros e outros acendedores (exceto os da p...   
9442  Cachimbos (incluindo os seus fornilhos), pitei...   
9443  Vaporizadores de toucador, suas armações e cab...   
9444  Garrafas térmicas e outros recipientes isotérm...   
9445  Manequins e artigos semelhantes; autômatos e c...   

                                             PRODUTOSH2         CLASSIFICACAO  \
1     Gorduras e óleos animais ou vegetais; produtos...  Baseados em Recursos   
2                         Café, chá, mate e especiarias    Produtos Primários   
3     Sementes e frutos oleaginosos; grãos, sementes...    Produtos Primários   
4                         Café, chá, mate e especiarias    Produtos Primários   
5                         Café, chá, mate e especiarias    Produtos Primários   
...                                                 ...                   ...   
9441                                     Obras diversas      Baixa Tecnologia   
9442                                     Obras diversas      Baixa Tecnologia   
9443                                     Obras diversas      Baixa Tecnologia   
9444                                     Obras diversas      Baixa Tecnologia   
9445                                     Obras diversas      Baixa Tecnologia   

          EXP       ECI  VCR  ...  OBITO  ISOLAMENTO  MONITORAMENTO  \
1      465320  0.021937    1  ...      3           1              0   
2       11500 -0.021426    1  ...      3           1              0   
3      435598 -0.021426    1  ...      3           1              0   
4      250621 -0.511772    1  ...      3           1              0   
5     4351923  0.020993    1  ...      3           1              0   
...       ...       ...  ...  ...    ...         ...            ...   
9441      332  0.020993    0  ...    722           1              1   
9442       45  0.020993    0  ...    722           1              1   
9443      230  0.020993    0  ...    722           1              1   
9444       95  0.020993    0  ...    722           1              1   
9445      375  0.020993    0  ...    722           1              1   

      BARREIRAS SANCOES  TENDAS  SOBRECARGA  AMPLIACAOLEITE  HOSPITALCAMPANHA  \
1             1       1       1           0               1                 0   
2             1       1       1           0               1                 0   
3             1       1       1           0               1                 0   
4             1       1       1           0               1                 0   
5             1       1       1           0               1                 0   
...         ...     ...     ...         ...             ...               ...   
9441          1       1       0           0               0                 0   
9442          1       1       0           0               0                 0   
9443          1       1       0           0               0                 0   
9444          1       1       0           0               0                 0   
9445          1       1       0         

## 3.1.2.Dummy de monitoramento

Precisamos investigar quais municípios não tiveram informações que indicam presença (ou não) de monitoramento do isolamento social. Vale ressaltar que isto não significa que o município não teve monitoramento (o que é indicado pelo valor 0 na dummy), mas sim que não podemos afirmar se houve ou não monitoramento. Logo, para a análise desta política e seu impacto sobre número de mortes, temos que desconsiderar tais municípios.

In [ ]:
df_analise_missing_monitoramento = df[(df['MONITORAMENTO'] == 'NÃO DISPONÍVEL')].copy()
df_analise_missing_monitoramento.groupby(by=['MUNICIPIO']).count()

CODMUN  SH4  ANO  PRODUTOSH4  PRODUTOSH2  CLASSIFICACAO  \
MUNICIPIO                                                                     
BOM JESUS DO NORTE       3    3    3           3           3              3   

                    EXP  ECI  VCR  PIBpc  ...  OBITO  ISOLAMENTO  \
MUNICIPIO                                 ...                      
BOM JESUS DO NORTE    3    3    3      3  ...      3           3   

                    MONITORAMENTO  BARREIRAS  SANCOES  TENDAS  SOBRECARGA  \
MUNICIPIO                                                                   
BOM JESUS DO NORTE              3          3        3       3           3   

                    AMPLIACAOLEITE  HOSPITALCAMPANHA  FAIXA_POP_2020  
MUNICIPIO                                                             
BOM JESUS DO NORTE               3                 3               3  

[1 rows x 21 columns]

Vemos que só o município de Bom Jesus do Norte não disponibilizou tal informação. Logo, este município não irá agregar, no geral, informações para o nosso estudo.

In [ ]:
df_analise = df_analise[(df_analise['MONITORAMENTO'] != 'NÃO DISPONÍVEL')].copy()
df_analise

CODMUN   SH4   ANO       MUNICIPIO  \
1     3200102  1507  2017  AFONSO CLAUDIO   
2     3200102   901  2018  AFONSO CLAUDIO   
3     3200102  1201  2018  AFONSO CLAUDIO   
4     3200102   901  2019  AFONSO CLAUDIO   
5     3200102   901  2020  AFONSO CLAUDIO   
...       ...   ...   ...             ...   
9441  3205309  9613  2020         VITORIA   
9442  3205309  9614  2020         VITORIA   
9443  3205309  9616  2020         VITORIA   
9444  3205309  9617  2020         VITORIA   
9445  3205309  9618  2020         VITORIA   

                                             PRODUTOSH4  \
1     Óleo de soja e respectivas fracções, mesmo ref...   
2     Café, mesmo torrado ou descafeinado; cascas e ...   
3                                 Soja, mesmo triturada   
4     Café, mesmo torrado ou descafeinado; cascas e ...   
5     Café, mesmo torrado ou descafeinado; cascas e ...   
...                                                 ...   
9441  Isqueiros e outros acendedores (exceto os da p...   
9442  Cachimbos (incluindo os seus fornilhos), pitei...   
9443  Vaporizadores de toucador, suas armações e cab...   
9444  Garrafas térmicas e outros recipientes isotérm...   
9445  Manequins e artigos semelhantes; autômatos e c...   

                                             PRODUTOSH2         CLASSIFICACAO  \
1     Gorduras e óleos animais ou vegetais; produtos...  Baseados em Recursos   
2                         Café, chá, mate e especiarias    Produtos Primários   
3     Sementes e frutos oleaginosos; grãos, sementes...    Produtos Primários   
4                         Café, chá, mate e especiarias    Produtos Primários   
5                         Café, chá, mate e especiarias    Produtos Primários   
...                                                 ...                   ...   
9441                                     Obras diversas      Baixa Tecnologia   
9442                                     Obras diversas      Baixa Tecnologia   
9443                                     Obras diversas      Baixa Tecnologia   
9444                                     Obras diversas      Baixa Tecnologia   
9445                                     Obras diversas      Baixa Tecnologia   

          EXP       ECI  VCR  ...  OBITO  ISOLAMENTO  MONITORAMENTO  \
1      465320  0.021937    1  ...      3           1              0   
2       11500 -0.021426    1  ...      3           1              0   
3      435598 -0.021426    1  ...      3           1              0   
4      250621 -0.511772    1  ...      3           1              0   
5     4351923  0.020993    1  ...      3           1              0   
...       ...       ...  ...  ...    ...         ...            ...   
9441      332  0.020993    0  ...    722           1              1   
9442       45  0.020993    0  ...    722           1              1   
9443      230  0.020993    0  ...    722           1              1   
9444       95  0.020993    0  ...    722           1              1   
9445      375  0.020993    0  ...    722           1              1   

      BARREIRAS SANCOES  TENDAS  SOBRECARGA  AMPLIACAOLEITE  HOSPITALCAMPANHA  \
1             1       1       1           0               1                 0   
2             1       1       1           0               1                 0   
3             1       1       1           0               1                 0   
4             1       1       1           0               1                 0   
5             1       1       1           0               1                 0   
...         ...     ...     ...         ...             ...               ...   
9441          1       1       0           0               0                 0   
9442          1       1       0           0               0                 0   
9443          1       1       0           0               0                 0   
9444          1       1       0           0               0                 0   
9445          1       1       0         

In [ ]:
df_analise.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6163 entries, 1 to 9445
Data columns (total 22 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   CODMUN            6163 non-null   object 
 1   SH4               6163 non-null   object 
 2   ANO               6163 non-null   int64  
 3   MUNICIPIO         6163 non-null   object 
 4   PRODUTOSH4        6163 non-null   object 
 5   PRODUTOSH2        6163 non-null   object 
 6   CLASSIFICACAO     6163 non-null   object 
 7   EXP               6163 non-null   int64  
 8   ECI               6163 non-null   float64
 9   VCR               6163 non-null   int64  
 10  PIBpc             6163 non-null   float64
 11  CresPIBpc         6163 non-null   float64
 12  OBITO             6163 non-null   int64  
 13  ISOLAMENTO        6163 non-null   int64  
 14  MONITORAMENTO     6163 non-null   object 
 15  BARREIRAS         6163 non-null   int64  
 16  SANCOES           6163 non-null   int64  


In [ ]:
df_analise['MONITORAMENTO'] = df_analise['MONITORAMENTO'].astype(int)
df_analise.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6163 entries, 1 to 9445
Data columns (total 22 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   CODMUN            6163 non-null   object 
 1   SH4               6163 non-null   object 
 2   ANO               6163 non-null   int64  
 3   MUNICIPIO         6163 non-null   object 
 4   PRODUTOSH4        6163 non-null   object 
 5   PRODUTOSH2        6163 non-null   object 
 6   CLASSIFICACAO     6163 non-null   object 
 7   EXP               6163 non-null   int64  
 8   ECI               6163 non-null   float64
 9   VCR               6163 non-null   int64  
 10  PIBpc             6163 non-null   float64
 11  CresPIBpc         6163 non-null   float64
 12  OBITO             6163 non-null   int64  
 13  ISOLAMENTO        6163 non-null   int64  
 14  MONITORAMENTO     6163 non-null   int64  
 15  BARREIRAS         6163 non-null   int64  
 16  SANCOES           6163 non-null   int64  


## 3.2.Análise dataframe

### 3.2.1.PIB per capita (nível e crescimento) e nível exportado

Vemos, a partir da tabela abaixo, que:

(i) Os anos que, na média, o PIB per capita do Espírito Santo cresceu, em comparação com o ano anterior, foram 2018 e 2020. O caso de 2020, a priori, é curioso, pois, em tese e por hipótese, com a pandemia houve uma retração da atividade econômica. Todavia, também por hipótese, isso pode ser fruto das medidas fiscais expansionistas, tal qual o auxílio emergencial. Quando avaliamos a mediana, vemos que, em 2019, mais de 50% dos municípios do Espírito Santo tiveram crescimento do PIB per capita, o que indica para valores extremos no limite inferior da distribuição.

(ii) O patamar do PIB per capita aumentou entre 2017 e 2018, saindo de RS 31.000,00 para a casa dos RS 40.000,00. Porém, analisando a mediana, vemos que o valor diminui consideravelmente para todos os anos, exceto 2015, indicando para um número muito maior de municípios com PIB per capita abaixo de RS 30.000,00.

In [ ]:
df_analise.pivot_table(index = 'ANO',
                values = ['PIBpc', 'CresPIBpc', 'EXP'],
                aggfunc = ['median', 'mean']).round(4)

median                         mean                          
     CresPIBpc      EXP     PIBpc CresPIBpc           EXP       PIBpc
ANO                                                                  
2016   -0.0070  10150.0  27523.40   -0.0220  7.607059e+06  33525.0506
2017   -0.0028   9787.5  26766.38   -0.0003  9.660972e+06  31503.5711
2018    0.2023   2339.5  28523.85    0.1883  6.733046e+06  41850.1695
2019    0.0091   2703.0  26129.30   -0.0092  4.386488e+06  40330.4733
2020    0.0193   2175.0  29612.66    0.0283  3.121538e+06  41648.7354

### 3.2.2.Complexidade econômica

Outra análise importante a ser feita neste momento é a classificação dos municípios por nível de complexidade. Iremos analisar a complexidade para o ano de 2019. Por que? O nível de complexidade da estrutura produtiva de 2019 é o que antecede a pandemia, ou seja, é o mais próximo que temos para relacionar o estado da estrutura produtiva dos municípios no momento da pandemia.

In [ ]:
df_complexidade = df_analise[(df_analise['ANO'] == 2019)].copy()
df_complexidade

CODMUN   SH4   ANO       MUNICIPIO  \
4     3200102   901  2019  AFONSO CLAUDIO   
21    3200136   901  2019    AGUIA BRANCA   
22    3200136  2506  2019    AGUIA BRANCA   
23    3200136  2516  2019    AGUIA BRANCA   
24    3200136  6802  2019    AGUIA BRANCA   
...       ...   ...   ...             ...   
8973  3205309  9609  2019         VITORIA   
8974  3205309  9613  2019         VITORIA   
8975  3205309  9617  2019         VITORIA   
8976  3205309  9705  2019         VITORIA   
8977  3205309  9998  2019         VITORIA   

                                             PRODUTOSH4  \
4     Café, mesmo torrado ou descafeinado; cascas e ...   
21    Café, mesmo torrado ou descafeinado; cascas e ...   
22    Quartzo (exceto areias naturais); quartzites, ...   
23    Granito, pórfiro, basalto, arenito e outras pe...   
24    Pedras de cantaria ou de construção (exceto de...   
...                                                 ...   
8973  Lápis (exceto os da posição 9608), minas, past...   
8974  Isqueiros e outros acendedores (exceto os da p...   
8975  Garrafas térmicas e outros recipientes isotérm...   
8976  Coleções e espécimes para coleções, de zoologi...   
8977  Consumo de bordo (combustíveis e lubrificantes...   

                                             PRODUTOSH2         CLASSIFICACAO  \
4                         Café, chá, mate e especiarias    Produtos Primários   
21                        Café, chá, mate e especiarias    Produtos Primários   
22    Sal; enxofre; terras e pedras; gesso, cal e ci...    Produtos Primários   
23    Sal; enxofre; terras e pedras; gesso, cal e ci...    Produtos Primários   
24    Obras de pedra, gesso, cimento, amianto, mica ...  Baseados em Recursos   
...                                                 ...                   ...   
8973                                     Obras diversas      Baixa Tecnologia   
8974                                     Obras diversas      Baixa Tecnologia   
8975                                     Obras diversas      Baixa Tecnologia   
8976         Objetos de arte, de coleção e antiguidades                Outros   
8977                               Transações especiais                Outros   

          EXP       ECI  VCR  ...  OBITO  ISOLAMENTO  MONITORAMENTO  \
4      250621 -0.511772    1  ...      3           1              0   
21    1561570 -0.139122    1  ...      4           1              1   
22      10086 -0.139122    1  ...      4           1              1   
23    1715668 -0.139122    1  ...      4           1              1   
24    1938129 -0.139122    1  ...      4           1              1   
...       ...       ...  ...  ...    ...         ...            ...   
8973       27 -0.021706    0  ...    722           1              1   
8974       37 -0.021706    0  ...    722           1              1   
8975      146 -0.021706    0  ...    722           1              1   
8976     5000 -0.021706    1  ...    722           1              1   
8977     2703 -0.021706    0  ...    722           1              1   

      BARREIRAS  SANCOES  TENDAS  SOBRECARGA  AMPLIACAOLEITE  \
4             1        1       1           0               1   
21            1        1       1           0               1   
22            1        1       1           0               1   
23            1        1       1           0               1   
24            1        1       1           0               1   
...         ...      ...     ...         ...             ...   
8973          1        1       0           0               0   
8974          1        1       0           0               0   
8975          1        1       0           0               0   
8976          1        1       0           0               0   
8977          1        1       0           0               0   

      HOSPITALCAMPANHA         FAIXA_POP_2020  
4                    0    4 - 20001 até 50000  
21                   0     2 - 5001 até 10000  
22                

In [ ]:
df_analisecomplex = df_complexidade.pivot_table(index = ['MUNICIPIO'],
                values = 'ECI',
                aggfunc = 'mean'
                ).round(4)

Abaixo, vemos os top 10 Municípios mais complexos do Espírito Santo, Por hipótese, estes municípios apresentam melhor capacidade de resposta à pandemia, pois, quanto maior o nível de conhecimento embutido na estrutura produtiva (medido pelo ECI), maior a capacidade de resposta ao choque exógeno de saúde.

In [ ]:
df_analisecomplex.sort_values(['ECI'],ascending=False).head(10)

ECI
MUNICIPIO                     
COLATINA                0.3066
LINHARES                0.2643
GUARAPARI               0.2500
SERRA                   0.2011
CASTELO                 0.1795
VILA VELHA              0.1549
ANCHIETA                0.1513
BARRA DE SAO FRANCISCO  0.1324
SAO ROQUE DO CANAA      0.1324
SANTA MARIA DE JETIBA   0.1321

### 3.2.3.Diversificação da estrutura produtiva

Primeiro, deixaremos uma base de dados somente com informações de diversificação por classificação tecnológica. Esta base terá a informação da competitividade do munícipio por ano em cada classificação tecnológica, isto é, quantos produtos ele exporta competitivamente em comparação com o restante do país.

In [ ]:
df_diversificacao = df[['ANO', 'MUNICIPIO', 'CLASSIFICACAO', 'VCR', 'ECI']].copy()
df_diversificacao

ANO       MUNICIPIO         CLASSIFICACAO  VCR       ECI
0      2015  AFONSO CLAUDIO    Produtos Primários    1 -0.289533
1      2017  AFONSO CLAUDIO  Baseados em Recursos    1  0.021937
2      2018  AFONSO CLAUDIO    Produtos Primários    1 -0.021426
3      2018  AFONSO CLAUDIO    Produtos Primários    1 -0.021426
4      2019  AFONSO CLAUDIO    Produtos Primários    1 -0.511772
...     ...             ...                   ...  ...       ...
10464  2022         VITORIA      Baixa Tecnologia    0  0.020391
10465  2022         VITORIA      Baixa Tecnologia    0  0.020391
10466  2022         VITORIA                Outros    0  0.020391
10467  2022         VITORIA                Outros    0  0.020391
10468  2022         VITORIA                Outros    0  0.020391

[10469 rows x 5 columns]

In [ ]:
df_diversificacao_analise = df_diversificacao.pivot_table(index = ['ANO', 'MUNICIPIO'],
                values = 'VCR',
                columns='CLASSIFICACAO',
                aggfunc = 'sum'
                ).round(2).copy()

In [ ]:
df_diversificacao_analise.fillna(0)

CLASSIFICACAO            Alta Tecnologia  Baixa Tecnologia  \
ANO  MUNICIPIO                                               
2015 AFONSO CLAUDIO                  0.0               0.0   
     AGUA DOCE DO NORTE              0.0               0.0   
     AGUIA BRANCA                    0.0               0.0   
     ANCHIETA                        0.0               0.0   
     ARACRUZ                         0.0               1.0   
...                                  ...               ...   
2022 VIANA                           0.0               3.0   
     VILA PAVAO                      0.0               0.0   
     VILA VALERIO                    0.0               1.0   
     VILA VELHA                      1.0              14.0   
     VITORIA                         2.0               9.0   

CLASSIFICACAO            Baseados em Recursos  Média Tecnologia  Outros  \
ANO  MUNICIPIO                                                            
2015 AFONSO CLAUDIO                       0.0               0.0     0.0   
     AGUA DOCE DO NORTE                   1.0               0.0     0.0   
     AGUIA BRANCA                         1.0               0.0     0.0   
     ANCHIETA                             1.0               1.0     0.0   
     ARACRUZ                              2.0               0.0     0.0   
...                                       ...               ...     ...   
2022 VIANA                                0.0               2.0     0.0   
     VILA PAVAO                           0.0               0.0     0.0   
     VILA VALERIO                         0.0               0.0     0.0   
     VILA VELHA                          28.0               9.0     1.0   
     VITORIA                             15.0               2.0     1.0   

CLASSIFICACAO            Produtos Primários  
ANO  MUNICIPIO                               
2015 AFONSO CLAUDIO                     1.0  
     AGUA DOCE DO NORTE                 1.0  
     AGUIA BRANCA                       1.0  
     ANCHIETA                           0.0  
     ARACRUZ                            0.0  
...                                     ...  
2022 VIANA                              4.0  
     VILA PAVAO                         1.0  
     VILA VALERIO                       3.0  
     VILA VELHA                        30.0  
     VITORIA                           27.0  

[418 rows x 6 columns]

No geral, vemos que o estado do Espírito Santo é competitivo, principalmente em produtos primários e de baixa tecnologia. Além disso, entre 2015 e 2021, o estado parece ter elevado sua competitividade de modo geral, aumentando o número de produtos que exportados com vantagem comparativa revelada nos setores de alta tecnologia (6 para 18), baixa tecnologia (47 para 107), produtos primários (103 para 201) e produtos baseados em recursos (44 para 93). No setor de média tecnologia, os valores cresceram de 35 para 52 entre 2015 e 2016, mas, posterior a esse fato, não há tendência específica, alternando entre crescimento e decrescimento.

In [ ]:
df_diversificacao.pivot_table(index = 'ANO',
                values = 'VCR',
                columns='CLASSIFICACAO',
                aggfunc = 'sum'
                ).round(2)

CLASSIFICACAO  Alta Tecnologia  Baixa Tecnologia  Baseados em Recursos  \
ANO                                                                      
2015                         6                47                    44   
2016                         8                56                    44   
2017                         8                75                    55   
2018                         2                63                    63   
2019                         2                82                    86   
2020                        13                87                    95   
2021                        18               107                    93   
2022                         9               100                    93   

CLASSIFICACAO  Média Tecnologia  Outros  Produtos Primários  
ANO                                                          
2015                         35       2                 103  
2016                         52       3                 112  
2017                         56       2                 125  
2018                         41       4                 180  
2019                         49       5                 194  
2020                         52       5                 184  
2021                         48       6                 201  
2022                         47       5                 210

### 3.2.4.Número de óbitos

O número de óbitos é medido exclusivamente para o ano de 2020 (entre março e dezembro). Abaixo, vemos os municípios que mais registraram óbitos por COVID-19. Os três municípios que registraram mais óbitos foram Serra, Vitoria e Vila Velha.

Este resultado levanta a questão: estes municípios registraram mais óbitos muito em função de sua maior população ou por que tiveram pior desempenho no combate ao COVID-19?

Vemos que Vila Velha e Guarapari adotaram as medidas de distanciamento social apenas. Assim, não tiveram ampliação dos leitos ou instalação de hospital de campanha, além de observarem sobrecarga do sistema de saúde. Porém, Vila Velha teve 5,5 vezes mais óbitos. Dessa forma, conclui-se que talvez existe algo relacionado tamanho do município.

Este resultado indica que, quando formos analisar a relação entre complexidade, capacidade de resposta à pandemia e número de óbitos, talvez tenhamos que controlar pelo tamanho da população. Isto é, entre o grupo de municípios do mesmo tamanho, os mais complexos se saíram melhor na pandemia?


In [ ]:
df_obito = df_analise[(df_analise['ANO'] == 2020)].copy()
df_obito_2 = df_obito.pivot_table(index = 'MUNICIPIO',
                values = ['OBITO', 'ISOLAMENTO', 'MONITORAMENTO', 'BARREIRAS', 'SANCOES', 'TENDAS', 'SOBRECARGA', 'AMPLIACAOLEITE', 'HOSPITALCAMPANHA'],
                aggfunc = 'mean'
                ).sort_values('OBITO',ascending=False)
df_obito_2

AMPLIACAOLEITE  BARREIRAS  HOSPITALCAMPANHA  \
MUNICIPIO                                                              
SERRA                                 1          0                 0   
VITORIA                               0          1                 0   
VILA VELHA                            0          1                 0   
LINHARES                              1          0                 0   
CACHOEIRO DE ITAPEMIRIM               0          1                 0   
COLATINA                              1          0                 0   
SAO MATEUS                            1          1                 0   
CARIACICA                             1          1                 0   
ARACRUZ                               1          1                 0   
GUARAPARI                             0          1                 0   
SANTA TERESA                          1          1                 0   
BARRA DE SAO FRANCISCO                1          1                 0   
CASTELO                               0          0                 0   
ITAPEMIRIM                            1          1                 0   
IRUPI                                 0          1                 0   
IBIRACU                               1          1                 0   
JOAO NEIVA                            0          1                 0   
MARATAIZES                            0          1                 0   
SAO GABRIEL DA PALHA                  0          1                 0   
BAIXO GUANDU                          1          1                 0   
JERONIMO MONTEIRO                     1          0                 0   
GUACUI                                1          1                 0   
SOORETAMA                             0          1                 0   
MARILANDIA                            0          1                 0   
MUNIZ FREIRE                          1          1                 0   
NOVA VENECIA                          0          1                 0   
RIO BANANAL                           1          1                 0   
CONCEICAO DO CASTELO                  0          1                 0   
BOA ESPERANCA                         0          1                 0   
DOMINGOS MARTINS                      1          1                 0   
PANCAS                                1          1                 0   
ANCHIETA                              1          1                 0   
PINHEIROS                             1          1                 1   
RIO NOVO DO SUL                       0          1                 0   
VILA VALERIO                          0          1                 0   
AGUIA BRANCA                          1          1                 0   
VARGEM ALTA                           0          1                 0   
VIANA                                 1          1                 0   
VILA PAVAO                            0          1                 0   
AFONSO CLAUDIO                        1          1                 0   
ECOPORANGA                            0          1                 0   
DORES DO RIO PRETO                    0          1                 0   
PIUMA                                 1          1                 0   
SAO DOMINGOS DO NORTE                 0          1                 0   
SAO ROQUE DO CANAA                    0          0                 0   
ALFREDO CHAVES                        0          1                 0   
JAGUARE                               0          1                 0   
VENDA NOVA DO IMIGRANTE               0          1                 0   
ATILIO VIVACQUA                       1          1                 0   
MARECHAL FLORIANO                     0          1                 0   
SANTA LEOPOLDINA                      1          1                 0   
SANTA MARIA DE JETIBA                 0          1                 0   
MIMOSO DO SUL                         1          1                 0   
AGUA DOCE DO NORTE                    0          1               

# 4.Visualização dos dados

## 4.1.Diversificação produtiva

### 4.1.1.Espírito Santo

Nessa seção, queremos verificar a evolução da diversificação produtiva do estado do Espírito Santo.

In [ ]:
df_diversificacao_analise_2 = df_diversificacao.pivot_table(index = ['ANO', 'MUNICIPIO', 'CLASSIFICACAO'],
                values = 'VCR',
                aggfunc = 'sum'
                ).round(2)
df_diversificacao_analise_2.reset_index(inplace=True)

In [ ]:
df_diversificacao_analise_3 = df_diversificacao.pivot_table(index = ['ANO', 'CLASSIFICACAO'],
                values = 'VCR',
                aggfunc = 'sum'
                ).round(2)
df_diversificacao_analise_3.reset_index(inplace=True)

O gráfico deixa claro que o Espírito Santo reforçou, no geral, uma tendência de reprimarização da estrutura produtiva. O crescimento de mais de 100% no número de produtos primários com vantagem comparativa revelada reflete esta tendência.

In [ ]:
fig1 = px.line(df_diversificacao_analise_3, x='ANO', y='VCR', color='CLASSIFICACAO',
              markers=True)
fig1

### 4.1.2.Municípios

Além disso, como queremos relacionar complexidade econômica, capacidade de resposta dos municípios ao COVID-19 e número de mortes, vamos verificar a evolução da estrutura produtiva dos munícipios mais complexos, dos municípios que mais registraram óbitos e dos municípios que, em tese, tomaram mais medidas contra o COVID-19.

In [ ]:
df_diversificacao_analise_4 = df_diversificacao_analise_2[(df_diversificacao_analise_2['MUNICIPIO'] == 'COLATINA') |
                                                          (df_diversificacao_analise_2['MUNICIPIO'] == 'LINHARES') |
                                                          (df_diversificacao_analise_2['MUNICIPIO'] == 'GUARAPARI') |
                                                          (df_diversificacao_analise_2['MUNICIPIO'] == 'SERRA') |
                                                          (df_diversificacao_analise_2['MUNICIPIO'] == 'CASTELO')].copy()

In [ ]:
df_diversificacao_analise_5 = df_diversificacao_analise_4.pivot_table(index = ['ANO', 'CLASSIFICACAO', 'MUNICIPIO'],
                values = 'VCR',
                aggfunc = 'sum'
                ).round(2)
df_diversificacao_analise_5.reset_index(inplace=True)

In [ ]:
df_analisecomplex.sort_values(['ECI'],ascending=False).head(5)

ECI
MUNICIPIO        
COLATINA   0.3066
LINHARES   0.2643
GUARAPARI  0.2500
SERRA      0.2011
CASTELO    0.1795

Algumas observações da análise de diversificação dos top 5 municípios em complexidade.

(i) Apesar de ser somente o quarto município na classificação por em complexidade, Serra é mais competitiva em produtos de média tecnologia, seguida por Guarapari, que é o segundo. Em termos de alta complexidade, comparativamente em relação aos demais municípios, Guarapari e Serra são os mais competitivos.

(ii) Colatina, que é o município mais complexo, é mais competitiva em produtos de baixa tecnologia.

(iii) Serra é, claramente, o município com competitividade em um maior número de setores. Isso nos leva a indagar: Por que o município é, então, somente o quarto em complexidade?

Talvez os produtos em que Serra é competitiva são exportados competitivamente por uma diversidade maior de municípios no Brasil e/ou estes municípios são menos complexos, no geral. Todavia, esta é uma análise geral para o período 2016 - 2020. Temos que analisar somente o ano de 2019 (que é a base para classificar os municípios).

In [ ]:
fig2 = px.bar(df_diversificacao_analise_5, x = 'CLASSIFICACAO', y = 'VCR', color = 'MUNICIPIO',
              barmode='group')
fig2.update_traces(marker_line_width=1.5,
                  marker_line_color='black',
                  opacity=0.7)
fig2

In [ ]:
df_diversificacao_analise_6 = df_diversificacao_analise_5[(df_diversificacao_analise_5['ANO'] == 2019)].copy()

Mesmo analisando somente 2019, Serra e Guarapari são mais competitivas em mais setores, principalmente nos setores de Baixa Tecnologia (Guarapari é top 1) e Média Tecnologia (Serra é top 1 com folga). Mas, conforme adiantamos, complexidade econômica é uma medida iterativa e leva em consideração o Brasil inteiro, não somente o cenário do Espírito Santo. Por isso, saber em quais produtos estes municípios são competitivos e qual o nível de complexidade dos municípios brasileiros que são competitivos no mesmo produto é importante.

In [ ]:
fig3 = px.bar(df_diversificacao_analise_6, x = 'CLASSIFICACAO', y = 'VCR', color = 'MUNICIPIO',
             barmode='group')
fig3.update_traces(marker_line_width=1.5,
                  marker_line_color='black',
                  opacity=0.5)
fig3

Outro argumento que pode ir na direção contrária da lógica de "Serra é competitiva em mais bens de alta intensidade tecnológica, logo deveria ser mais complexa", é avaliar se existem municípios mais competitivos em produtos de média e baixa complexidade.

Vemos que Vila Velha é quase tão competitiva quanto Serra em produtos de média tecnologia, mas é somente sexta em complexidade. Cariacica aparece no top 5 de competitividade em produtos de alta, média e baixa complexidade, mas não é nem top 10 em complexidade. Ou seja, a análise não é tão direta.

In [ ]:
df_diversificacao_analise_7 = df_diversificacao_analise_2[(df_diversificacao_analise_2['ANO'] == 2019)].copy()
df_diversificacao_analise_7.sort_values(['CLASSIFICACAO','VCR'],ascending=False).groupby('CLASSIFICACAO').head(5)

ANO                MUNICIPIO         CLASSIFICACAO  VCR
655  2019               VILA VELHA    Produtos Primários   27
661  2019                  VITORIA    Produtos Primários   20
632  2019                    SERRA    Produtos Primários   11
541  2019  CACHOEIRO DE ITAPEMIRIM    Produtos Primários    8
589  2019                 LINHARES    Produtos Primários    8
546  2019                CARIACICA                Outros    2
660  2019                  VITORIA                Outros    2
631  2019                    SERRA                Outros    1
540  2019  CACHOEIRO DE ITAPEMIRIM                Outros    0
556  2019                 COLATINA                Outros    0
630  2019                    SERRA      Média Tecnologia   14
653  2019               VILA VELHA      Média Tecnologia   11
545  2019                CARIACICA      Média Tecnologia    8
568  2019                GUARAPARI      Média Tecnologia    5
646  2019                    VIANA      Média Tecnologia    2
652  2019               VILA VELHA  Baseados em Recursos   24
658  2019                  VITORIA  Baseados em Recursos   13
629  2019                    SERRA  Baseados em Recursos    6
538  2019  CACHOEIRO DE ITAPEMIRIM  Baseados em Recursos    5
526  2019          ATILIO VIVACQUA  Baseados em Recursos    3
543  2019                CARIACICA      Baixa Tecnologia   18
566  2019                GUARAPARI      Baixa Tecnologia    9
651  2019               VILA VELHA      Baixa Tecnologia    8
553  2019                 COLATINA      Baixa Tecnologia    7
590  2019             MANTENOPOLIS      Baixa Tecnologia    4
542  2019                CARIACICA       Alta Tecnologia    2
514  2019             AGUIA BRANCA       Alta Tecnologia    0
521  2019                  ARACRUZ       Alta Tecnologia    0
536  2019  CACHOEIRO DE ITAPEMIRIM       Alta Tecnologia    0
548  2019                  CASTELO       Alta Tecnologia    0

In [ ]:
fig4 = px.bar(df_diversificacao_analise_7, x = 'MUNICIPIO', y='VCR', color='CLASSIFICACAO')
fig4

## 4.2.Óbitos e tamanho do município

Os municípios mais complexos estão, de 56 municípios, no top-13 em número de óbitos.

Colatina: sexto em número de óbitos.

Linhares: quarto em número de óbitos.

Guarapari: décimo em número de óbitos.

Serra: primeiro em número de mortes.

Castelo: décimo terceiro em número de mortes.

Abaixo, analisamos a relação entre tamanho da população, média de complexidade e média do número de óbitos. Fica claro que municípios mais populosos são os mais complexos e o que registraram, na média, mais óbitos por COVID-19. Ou seja, há de fato um viés na análise. Para mitigar este viés na modelagem da relação entre complexidade e número de óbitos, iremos realizá-la para cada grupo populacional.

In [ ]:
df_obito_analise = df_analise[(df_analise['ANO'] == 2019)].copy()
df_obito_analise_2 = df_obito_analise.pivot_table(index = ['FAIXA_POP_2020'],
                values = ['OBITO','ECI'],
                aggfunc = 'mean'
                ).round(2)
df_obito_analise_2.reset_index(inplace=True)
df_obito_analise_2

FAIXA_POP_2020   ECI   OBITO
0     2 - 5001 até 10000 -0.12    2.88
1    3 - 10001 até 20000 -0.08    6.84
2    4 - 20001 até 50000  0.02   10.74
3   5 - 50001 até 100000  0.10    5.32
4  6 - 100001 até 500000  0.01  501.40
5   7 - Maior que 500000  0.17  752.27

In [ ]:
fig5 = px.scatter(df_obito_analise_2, x="ECI", y="OBITO", color="FAIXA_POP_2020")
fig5.update_traces(marker_size=7,
                  marker_line_width=1)
fig5

## 4.2.1.Capacidade de resposta

In [ ]:
df_resposta = df_analise[['MUNICIPIO', 'FAIXA_POP_2020', 'OBITO', 'ISOLAMENTO', 'MONITORAMENTO', 'BARREIRAS', 'SANCOES', 'TENDAS', 'SOBRECARGA',
                                'AMPLIACAOLEITE', 'HOSPITALCAMPANHA']].copy()

In [ ]:
df_resposta_2 = df_resposta.pivot_table(index = ['MUNICIPIO','FAIXA_POP_2020'],
                values = ['OBITO', 'ISOLAMENTO', 'MONITORAMENTO', 'BARREIRAS', 'SANCOES', 'TENDAS', 'SOBRECARGA', 'AMPLIACAOLEITE', 'HOSPITALCAMPANHA'],
                aggfunc = 'mean'
                ).round(2)
df_resposta_2.reset_index(inplace=True)

In [ ]:
df_resposta_3 = df_resposta_2.pivot_table(index = ['FAIXA_POP_2020'],
                values = ['ISOLAMENTO'],
                aggfunc = 'sum'
                ).round(2)
df_resposta_3.rename(columns = {'ISOLAMENTO' : 'QUANTIDADE'},
          inplace = True)
df_resposta_3.reset_index(inplace=True)
df_resposta_4 = df_resposta_2.pivot_table(index = ['FAIXA_POP_2020'],
                values = ['OBITO', 'ISOLAMENTO', 'MONITORAMENTO', 'BARREIRAS', 'SANCOES', 'TENDAS', 'SOBRECARGA', 'AMPLIACAOLEITE', 'HOSPITALCAMPANHA'],
                aggfunc = 'sum'
                ).round(2)
df_resposta_4.reset_index(inplace=True)

In [ ]:
df_merged = df_resposta_3.merge(df_resposta_4)
df_merged

FAIXA_POP_2020  QUANTIDADE  AMPLIACAOLEITE  BARREIRAS  \
0     2 - 5001 até 10000           5               2          5   
1    3 - 10001 até 20000          20               9         17   
2    4 - 20001 até 50000          23              13         20   
3   5 - 50001 até 100000           2               1          2   
4  6 - 100001 até 500000           8               5          6   
5   7 - Maior que 500000           2               1          1   

   HOSPITALCAMPANHA  ISOLAMENTO  MONITORAMENTO  OBITO  SANCOES  SOBRECARGA  \
0                 0           5              2     13        4           0   
1                 0          20             12    154       13           2   
2                 1          23             15    252       18           3   
3                 0           2              0     14        1           0   
4                 0           8              7   1584        7           5   
5                 0           2              2   2004        2           1   

   TENDAS  
0       3  
1       6  
2      12  
3       1  
4       4  
5       1

A coluna QUANTIDADE indica o número de municípios em cada categoria de faixa populacional. Apesar de existirem somente 2 municípios nas categoria 'Maior que 500000' e '50001 até 100000', elas registraram, no total, mais óbitos do que as demais, indicando que, de fato, o nível de óbitos é muito superior nas grandes cidades.

In [ ]:
df_merged["RAZAOAMPLIACAO"] = df_merged["AMPLIACAOLEITE"] / df_merged["QUANTIDADE"]
df_merged["RAZAOBARREIRAS"] = df_merged["BARREIRAS"] / df_merged["QUANTIDADE"]
df_merged["RAZAOHOSPITAL"] = df_merged["HOSPITALCAMPANHA"] / df_merged["QUANTIDADE"]
df_merged["RAZAOISOLAMENTO"] = df_merged["ISOLAMENTO"] / df_merged["QUANTIDADE"]
df_merged["RAZAOMONITORAMENTO"] = df_merged["MONITORAMENTO"] / df_merged["QUANTIDADE"]
df_merged["RAZAOSANCOES"] = df_merged["SANCOES"] / df_merged["QUANTIDADE"]
df_merged["RAZAOSOBRECARGA"] = df_merged["SOBRECARGA"] / df_merged["QUANTIDADE"]
df_merged["RAZAOTENDAS"] = df_merged["TENDAS"] / df_merged["QUANTIDADE"]
df_merged_2 = df_merged[['FAIXA_POP_2020', 'OBITO','RAZAOAMPLIACAO', 'RAZAOBARREIRAS', 'RAZAOHOSPITAL', 'RAZAOISOLAMENTO',
                         "RAZAOMONITORAMENTO", "RAZAOSANCOES", "RAZAOSOBRECARGA", "RAZAOTENDAS"]].head(5)
df_merged_2

FAIXA_POP_2020  OBITO  RAZAOAMPLIACAO  RAZAOBARREIRAS  \
0     2 - 5001 até 10000     13        0.400000        1.000000   
1    3 - 10001 até 20000    154        0.450000        0.850000   
2    4 - 20001 até 50000    252        0.565217        0.869565   
3   5 - 50001 até 100000     14        0.500000        1.000000   
4  6 - 100001 até 500000   1584        0.625000        0.750000   

   RAZAOHOSPITAL  RAZAOISOLAMENTO  RAZAOMONITORAMENTO  RAZAOSANCOES  \
0       0.000000              1.0            0.400000      0.800000   
1       0.000000              1.0            0.600000      0.650000   
2       0.043478              1.0            0.652174      0.782609   
3       0.000000              1.0            0.000000      0.500000   
4       0.000000              1.0            0.875000      0.875000   

   RAZAOSOBRECARGA  RAZAOTENDAS  
0         0.000000     0.600000  
1         0.100000     0.300000  
2         0.130435     0.521739  
3         0.000000     0.500000  
4         0.625000     0.500000

As faixas de população que registraram mais óbitos foram aquelas que também tiveram uma razão maior de municípios que ampliaram a quantidade de leitos.
Não se sabe a direção da causalidade, mas, por hipótese, pelo fato de estarem registrando mais óbitos, provavelmente também tiveram uma sobrecarga maior no sistema de saúde (o que é confirmado pela razão 0,625 para a faixa '100001 até 500000' e pela razão 0,130435 para a faixa '20001 até 50000'), logo, tiveram que ampliar os leitos.

O mesmo é observado para as variáveis que indicam a razão de municípios que realizaram monitoramento (razão 0,875 para a faixa '100001 até 500000' e pela razão 0,652174 para a faixa '20001 até 50000') e adotaram sanções (razão 0,875 para a faixa '100001 até 500000' e pela razão 0,782609 para a faixa '20001 até 50000'). Ou seja, em função das crescentes estatísticas, provavelmente os municípios tiveram que tomar atitudes para garantir o cumprimento das medidas de contenção do avanço do vírus.

# 5.Métodos Quantitativos

In [ ]:
import statsmodels.formula.api as smf

# 5.1.Óbitos vs. Complexidade Econômica

In [ ]:
df_analise_m1 = df_analise[(df_analise['ANO'] == 2019)].copy()
df_analise_m1 = df_analise_m1.pivot_table(index = ["MUNICIPIO", 'FAIXA_POP_2020'],
                values = ['OBITO','ECI','ISOLAMENTO', 'MONITORAMENTO', 'BARREIRAS', 'SANCOES', 'TENDAS', 'SOBRECARGA', 'AMPLIACAOLEITE', 'HOSPITALCAMPANHA'],
                aggfunc = 'mean'
                ).round(2)
df_analise_m1.reset_index(inplace=True)
df_analise_m1

MUNICIPIO         FAIXA_POP_2020  AMPLIACAOLEITE  BARREIRAS  \
0            AFONSO CLAUDIO    4 - 20001 até 50000               1          1   
1        AGUA DOCE DO NORTE    3 - 10001 até 20000               0          1   
2              AGUIA BRANCA     2 - 5001 até 10000               1          1   
3            ALFREDO CHAVES    3 - 10001 até 20000               0          1   
4                  ANCHIETA    4 - 20001 até 50000               1          1   
5                   ARACRUZ  6 - 100001 até 500000               1          1   
6           ATILIO VIVACQUA    3 - 10001 até 20000               1          1   
7              BAIXO GUANDU    4 - 20001 até 50000               1          1   
8    BARRA DE SAO FRANCISCO    4 - 20001 até 50000               1          1   
9             BOA ESPERANCA    3 - 10001 até 20000               0          1   
10  CACHOEIRO DE ITAPEMIRIM  6 - 100001 até 500000               0          1   
11                CARIACICA  6 - 100001 até 500000               1          1   
12                  CASTELO    4 - 20001 até 50000               0          0   
13                 COLATINA  6 - 100001 até 500000               1          0   
14     CONCEICAO DO CASTELO    3 - 10001 até 20000               0          1   
15         DOMINGOS MARTINS    4 - 20001 até 50000               1          1   
16       DORES DO RIO PRETO     2 - 5001 até 10000               0          1   
17               ECOPORANGA    4 - 20001 até 50000               0          1   
18                   FUNDAO    4 - 20001 até 50000               0          0   
19                GUARAPARI  6 - 100001 até 500000               0          1   
20                  IBIRACU    3 - 10001 até 20000               1          1   
21                   ICONHA    3 - 10001 até 20000               1          0   
22                    IRUPI    3 - 10001 até 20000               0          1   
23               ITAPEMIRIM    4 - 20001 até 50000               1          1   
24                     IUNA    4 - 20001 até 50000               1          0   
25                  JAGUARE    4 - 20001 até 50000               0          1   
26        JERONIMO MONTEIRO    3 - 10001 até 20000               1          0   
27               JOAO NEIVA    3 - 10001 até 20000               0          1   
28                 LINHARES  6 - 100001 até 500000               1          0   
29             MANTENOPOLIS    3 - 10001 até 20000               1          1   
30               MARATAIZES    4 - 20001 até 50000               0          1   
31        MARECHAL FLORIANO    3 - 10001 até 20000               0          1   
32               MARILANDIA    3 - 10001 até 20000               0          1   
33            MIMOSO DO SUL    4 - 20001 até 50000               1          1   
34             MUNIZ FREIRE    3 - 10001 até 20000               1          1   
35             NOVA VENECIA   5 - 50001 até 100000               0          1   
36                PINHEIROS    4 - 20001 até 50000               1          1   
37                    PIUMA    4 - 20001 até 50000               1          1   
38              RIO BANANAL    3 - 10001 até 20000               1          1   
39          RIO NOVO DO SUL    3 - 10001 até 20000               0          1   
40         SANTA LEOPOLDINA    3 - 10001 até 20000               1          1   
41    SANTA MARIA DE JETIBA    4 - 20001 até 50000               0          1   
42             SANTA TERESA    4 - 20001 até 50000               1          1   
43    SAO DOMINGOS DO NORTE     2 - 5001 até 10000               0          1   
44     SAO GABRIEL DA PALHA    4 - 20001 até 50000               0          1   
45               SAO MATEUS  6 - 100001 até 500000               1          1   
46       SAO ROQUE DO CANAA    3 - 10001 até 20000               0          0   
47                    SERRA   7 - Maior que 500000               1          0   
48                SOORETAMA    4 - 20001 até 5000

In [ ]:
df_analise_m1['FAIXA_POP_2020'].unique()

array(['4 - 20001 até 50000', '3 - 10001 até 20000', '2 - 5001 até 10000',
       '6 - 100001 até 500000', '5 - 50001 até 100000',
       '7 - Maior que 500000'], dtype=object)

In [ ]:
df_analise_m1["LOGOBITOS"] = np.log(df_analise_m1["OBITO"])

In [ ]:
# Modelo que explica, no geral, número de óbitos por nível de complexidade não é significativo e possui baixo teor explicativo
reg1 = smf.ols(formula = 'LOGOBITOS ~ ECI', data = df_analise_m1)
results1 = reg1.fit()
print(results1.summary())

                            OLS Regression Results                            
Dep. Variable:              LOGOBITOS   R-squared:                       0.062
Model:                            OLS   Adj. R-squared:                  0.044
Method:                 Least Squares   F-statistic:                     3.554
Date:                Fri, 28 Apr 2023   Prob (F-statistic):             0.0648
Time:                        12:23:22   Log-Likelihood:                -108.74
No. Observations:                  56   AIC:                             221.5
Df Residuals:                      54   BIC:                             225.5
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      2.4208      0.234     10.327      0.0

Definir uam duumy para categoria populacional, na qual a categoria '4 - 20001 até 50000' é a referência, em função da maior quantidade de municípios.

In [ ]:
df_analise_m1.loc[df_analise_m1['FAIXA_POP_2020'] == '2 - 5001 até 10000', 'Dummy_Faixa2'] = 1
df_analise_m1.loc[df_analise_m1['FAIXA_POP_2020'] != '2 - 5001 até 10000', 'Dummy_Faixa2'] = 0

df_analise_m1.loc[df_analise_m1['FAIXA_POP_2020'] == '3 - 10001 até 20000', 'Dummy_Faixa3'] = 1
df_analise_m1.loc[df_analise_m1['FAIXA_POP_2020'] != '3 - 10001 até 20000', 'Dummy_Faixa3'] = 0

df_analise_m1.loc[df_analise_m1['FAIXA_POP_2020'] == '5 - 50001 até 100000', 'Dummy_Faixa5'] = 1
df_analise_m1.loc[df_analise_m1['FAIXA_POP_2020'] != '5 - 50001 até 100000', 'Dummy_Faixa5'] = 0

df_analise_m1.loc[df_analise_m1['FAIXA_POP_2020'] == '6 - 100001 até 500000', 'Dummy_Faixa6'] = 1
df_analise_m1.loc[df_analise_m1['FAIXA_POP_2020'] != '6 - 100001 até 500000', 'Dummy_Faixa6'] = 0

df_analise_m1.loc[df_analise_m1['FAIXA_POP_2020'] == '7 - Maior que 500000', 'Dummy_Faixa7'] = 1
df_analise_m1.loc[df_analise_m1['FAIXA_POP_2020'] != '7 - Maior que 500000', 'Dummy_Faixa7'] = 0



In [ ]:
# Modelo que explica, no geral, número de óbitos por nível de complexidade não é significativo e possui alto teor explicativo.
reg2 = smf.ols(formula = 'LOGOBITOS ~ ECI + Dummy_Faixa2 + Dummy_Faixa3 + Dummy_Faixa5 + Dummy_Faixa6 + Dummy_Faixa7', data = df_analise_m1)
results2 = reg2.fit()
print(results2.summary())

                            OLS Regression Results                            
Dep. Variable:              LOGOBITOS   R-squared:                       0.689
Model:                            OLS   Adj. R-squared:                  0.651
Method:                 Least Squares   F-statistic:                     18.09
Date:                Fri, 28 Apr 2023   Prob (F-statistic):           6.25e-11
Time:                        12:23:46   Log-Likelihood:                -77.833
No. Observations:                  56   AIC:                             169.7
Df Residuals:                      49   BIC:                             183.8
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
Intercept        1.7693      0.229      7.723   

# 5.2.Óbitos vs. Variáveis de Capacidade de Resposta

In [ ]:
# Modelo que explica, no geral, número de óbitos por nível de capacidade de resposta não é significativo e possui baixo teor explicativo.
reg3 = smf.ols(formula = 'LOGOBITOS ~ ECI + AMPLIACAOLEITE + BARREIRAS + MONITORAMENTO + SANCOES + SOBRECARGA + TENDAS', data = df_analise_m1)
results3 = reg3.fit()
print(results3.summary())

                            OLS Regression Results                            
Dep. Variable:              LOGOBITOS   R-squared:                       0.206
Model:                            OLS   Adj. R-squared:                  0.090
Method:                 Least Squares   F-statistic:                     1.779
Date:                Fri, 28 Apr 2023   Prob (F-statistic):              0.113
Time:                        12:34:06   Log-Likelihood:                -104.07
No. Observations:                  56   AIC:                             224.1
Df Residuals:                      48   BIC:                             240.3
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Intercept          2.0066      0.829      2.

# 6.Conclusões

(i) Nenhum dos modelos apresentou significância, indicando para um grande esforço de refinamento para análises futuras ou, de fato, uma ausência de significância na relação entre ECI, Número de Óbitos e Capacidade de Resposta da forma que medimos.

(ii) As análises descritivas reveleram que municípios mais complexos também são aqueles com maior número populacional.

(iii) Municípios com maior população foram aqueles que também registraram mais óbitos.